teacher.rb

```diff
@@ -0,0 +1,40 @@
+require './person'
+
+class Teacher < Person
+  attr_reader :specialization, :parent_permission
+
+  def initialize(age, name = 'Unknown', specialization = 'Math')
```

Comment:
Same here, please include `parent_permission = true` as a parameter for the `Teacher` class to make sure that the value for `parent_permission` is modifiable.

- For the `Teacher` class, the public method `can_use_services?` should be overridden so it always returns true. Therefore, using `is_of_age?` will throw an error since the method does not exist in the `Teacher` class. You can simply return `true` in the method.

```
  def can_use_services?
    true
  end
```